### PR TITLE
Feature/feedback section modal

### PR DESCRIPTION
**Pull request by RobsonOlv**
#### What is the purpose of this pull request?

Add the modal that will appear when the user selects a feedback button.

[Documented Here :book: ](https://www.notion.so/vtexhandbook/Adicionar-modal-do-feedback-section-em-paginas-de-API-Guides-99ccf10b85d64c48a0745586878fcc98)

#### What problem is this solving?

The feedback section stays static. When the user selects a feedback button, a modal should appear with a `textarea` where the user can give us their feedback explanation. This PR implements the modal structure and logic. But, the logic for collecting and storing feedback will be implemented in future PRs.

#### How should this be manually tested?

Go to [this page](https://deploy-preview-49--elated-hoover-5c29bf.netlify.app/docs/api-guides/clients) and try:
* Select a feedback button (**yes** or **no**) and check:
   * The modal appears in the correct location: If there is space between the selected button and the top of the page, the modal should appear below it, otherwise, at the bottom. And check the arrow direction and position.
   * The page background is darkened
   * The scrollbar is locked, the user cannot scroll the page while the modal is appearing
   * The selected button stays `active` while the modal is appearing
* Click **outside** of the modal and check:
    * The background returned to normal
    * The scrollbar is enabled and the page scroll stays at the same place
    * The selected feedback button has returned to the default state
* Click on the `SEND FEEDBACK` button inside the modal and check:
    * The background returned to normal
    * The scrollbar is enabled and the page scroll stays at the same place
    * The selected button and the feedback section text have changed to the "submitted" appearance 

#### Screenshots or example usage

https://user-images.githubusercontent.com/42784961/173897926-109dccfb-ee86-4da1-9c8c-9547e1ed3246.mp4

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Review by amandascm — APPROVED** [review]
LGTM ✨

**Review by Meyrillan — APPROVED** [review]
I noticed that while I'm changing the screen width, the modal is moving to the left; and the title color "Leave a comment (optional)" is #545454.

Captura de Tela 2022-06-20 às 13 54 28

File: src/components/feedback-modal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            sx={styles.textarea}
+            label=""
+            rows={7}
+          />
+          <Button
+            onClick={() => closeModal()}
+            sx={styles.button}
+            variant="secondary"
+          >
+            send feedback
```
Comment (leonardoAnjos16) [inline src/components/feedback-modal/index.tsx:92]: I think the "send feeback" and the "Leave a comment" texts should be in the `language.json` file, right?

**Review by leonardoAnjos16 — APPROVED** [review]
LGTM 💯

**Review by melissafalcao — APPROVED** [review]
LGTM 👍🏻

File: src/components/feedback-modal/index.tsx (hunk 1 above)
Comment (RobsonOlv) [reply src/components/feedback-modal/index.tsx]: I will change it

**Review by amandascm — APPROVED** [review]

**Review by Meyrillan — APPROVED** [review]

**Review by JoaoMontenegr0 — APPROVED** [review]
Ótimo!

**Pull request merged**